src/pages/edit_resource/contents_step/FileFields.tsx

```diff
@@ -64,13 +80,76 @@ const FileFields: FC<FileFieldsProps> = ({ uppy, setAllChangesSaved }) => {
     }
   };
 
+  useEffect(() => {
+    const setNewThumbnailAPICallAndFormikChange = async () => {
+      try {
+        setFileInputIsBusy(true);
+        if (newThumbnailContent) {
+          await setContentAsDefaultThumbnail(values.resource.identifier, newThumbnailContent.identifier);
+          let tobeDeletedIdentifier = '';
+          for (let i = 0; i < values.resource.contents.length; i++) {
+            if (values.resource.contents[i].identifier === newThumbnailContent.identifier) {
+              console.log('detected correct content identifier', values.resource.contents[i].identifier);
+              values.resource.contents[i].features.dlr_thumbnail_default = 'true';
+            } else if (
+              values.resource.contents[i].identifier !== newThumbnailContent.identifier &&
+              values.resource.contents[i].features.dlr_thumbnail_default === 'true' &&
+              values.resource.contents[i].features.dlr_content_master === 'false'
+            ) {
+              console.log('detected the need to delete', values.resource.contents[i].identifier);
+              tobeDeletedIdentifier = values.resource.contents[i].identifier;
+            } else {
+              values.resource.contents[i].features.dlr_thumbnail_default = 'false';
+              console.log('detected master content', values.resource.contents[i].identifier);
+            }
+          }
+
+          setShouldPollNewThumbnail(true);
+          await new Promise((r) => setTimeout(r, 2000));
+          setShouldPollNewThumbnail(false);
+          setFileInputIsBusy(false);
+          await new Promise((r) => setTimeout(r, 3000));
+          if (tobeDeletedIdentifier.length > 0) {
+            await deleteResourceContent(values.resource.identifier, tobeDeletedIdentifier);
+            values.resource.contents = values.resource.contents.filter(
+              (content) => content.identifier !== tobeDeletedIdentifier
+            );
+          }
+        }
+      } catch (error) {
+        console.log(error);
+      }
+    };
+
+    if (thumbnailUppy) {
+      if (!thumbnailUppy.hasUploadSuccessEventListener) {
+        thumbnailUppy.on('complete', () => {
+          if (newThumbnailContent) {
+            setNewThumbnailAPICallAndFormikChange().then(() => {
+              thumbnailUppy.reset();
```

Comment:
Jeg ser for meg at reset() er en del av problemet, men uten reset må brukeren trykke "Done" på dashboardet, lukke dashboardet og åpne det igjen for å få lov til å legge inn ekstra filer.